extension/src/popup/ducks/transactionSubmission.ts

```diff
@@ -605,6 +609,7 @@ const transactionSubmissionSlice = createSlice({
     });
     builder.addCase(getAccountBalances.fulfilled, (state, action) => {
       state.accountBalances = action.payload;
+      state.tokensWithNoBalance = []; // TODO
```

Comment:
Ah, yeah - so this was intended to cast a wider net. This was implemented to allow users to remove a contract ID that they no longer wanted in Freighter's UI. It could be because a) the contract expired or b) the user mistyped  a contract id and it can't be found. It would def be great to be able to delineate between expired contracts that existed and contracts that never existed


(example of how `tokensWithNoBalance` is used to show a bad contract id that a user may want to remove)
<img width="357" alt="Screenshot 2023-12-04 at 5 56 30 PM" src="https://github.com/stellar/freighter/assets/6789586/e50b561e-f8c7-43fa-88bc-cc02e55a4265">

Reply:
nice yeah makes sense. I think the move to the backend makes recreating the behavior exactly kind of tough for a few reasons -
We will now have access to balances even when the underlying contract has expired, due to indexing.
Mistyping should result in a few cases depending on how bad the mis-type is - if the string is a contract ID but does not have the same interface as a token contract, this will fail at index time which should mean that we can reject it when they enter it in(not true for existing ones exactly), or if the contract string does not parse to a contract ID, the API will 400 the request to index it.

So I think that more or less leaves us with handling only the case of truly having a 0 balance in order to suggest removing, do you agree? Also, to complicate things further, this ignores the fallback to rpc cases which would work as you have already written.
@piyalbasu

Reply:
updated to handle the missing balance case in 4de74c01ea630a69c161d4d90a4a77a96ae4f8d3. I tried to go for "ban-aid" solution which maybe isn't the cleanest but will hopefully be easiest to change once we move into handling this more explicitly with different states for expired entries, etc.